dealership/contracts.py

```diff
@@ -1,12 +1,49 @@
+from dealership.vehicles import Car, Truck, Motorcycle
+
 class Contract(object):
     pass
 
-
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
-
+        self.vehicle = vehicle
+        self.customer = customer
+        self.monthly_payments = monthly_payments
+        
+    def interest_rate(self):
+        interest_dict = {Car : 1.07, Motorcycle: 1.03, Truck : 1.11}
```

Comment:
I like the use of the dictionary for the interest and lease multipliers. Another way I saw a few others do implement it is buy adding a constant variable to the car, truck, and motorcycle class and calling it using self.vehicle.interest.